### ISSUE TITLE
[a11y]: DatePicker [Range With Calendar] - weekdays in the calendar bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15254]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
The dates in the Calendar is not programatically associated with the coresponding Weeks

Corresponding Week Day is not announced by the screen reader in Application mode while navigating through the Dates with Arrow keys in the Calendar Note: Same issue is found in iPhone VO as well.

### Recommended Fix
While navigating through the dates within the calendar, Screen reader is expected to announce the curresponding Week Name as well. Otherwise, it would be dificult for the user to corelate the info.

When the required information/instruction is not programatically associated, it can be affect the user ability to understand and accomplish A task

### WCAG 2.1 Violation
1.3.1 Info and Relationships

**Comment by heloiselui** [comment]
Checked with our accessibility team, and this issue still exists.
